### PR TITLE
Add libpng

### PR DESCRIPTION
**Pull request by ocefpaf**
Maybe this can help us with #461

### CODE REVIEW

File: recipes/libpng/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{% set version = "1.6.21" %}
```
Comment (ocefpaf) [inline recipes/libpng/meta.yaml]: The default channel version is `1.6.17`, but the ftp no longer has that version. However, it seems that both are `libpng16`, so we should be OK here.

**Comment by ocefpaf** [comment]
@kmuehlbauer can you cherry pick this commit on #461 to see if this fix the OS X build?

File: recipes/libpng/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/libpng/meta.yaml]: Not that I can pinpoint something that says this is ok yet, but the feeling from what I have read is that it is ok. So, I'm not worried about this.

**Comment by jakirkham** [comment]
Seems to build on Mac. Loving the idea of `libpng` entering the stack. Unfortunately, feedstocking is still a mess.

**Comment by kmuehlbauer** [comment]
> @kmuehlbauer can you cherry pick this commit on #461 to see if this fix the OS X build?

I do not get it, which changes should I incorporate in #461?

**Comment by ocefpaf** [comment]
> I do not get it, which changes should I incorporate in #461?

It should be the whole PR, but now that this got merged let's just wait for the feedstock and re-start the CIs there.

@pelson it seems that the time-out issue still exists. I see that `ordered-set` made it, but `libpng` did not.

**Pull request merged**